### DISCUSSION TITLE
[Feat]: scrollNext/scrollPrev support for number to scroll

### DISCUSSION THREAD

**Discussion by meirroth** [davidjerleke/embla-carousel#833] [Ideas]
### Which variants of Embla Carousel are relevant to this feature request?

- [X] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [X] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### Feature description

I propose the introduction of an optional parameter to the `scrollNext` and `scrollPrev` methods, allowing users to specify the number of slides to advance or retreat. For instance, `scrollNext({ scroll: 4 })` would move forward by 4 slides. The default behavior would remain as a single slide movement.

Additionally, I suggest enhancing the library by providing reactive properties `canScrollPrev` and `canScrollNext` as computed properties. This would eliminate the need to update variables with getters on `slidesInView`, simplifying the code and improving reactivity.

### Additional Context

As I transition from [vueper-slides](https://github.com/antoniandre/vueper-slides) to Embla Carousel in my Nuxt application, I've found the setup process to be largely seamless, barring the reactive direction issue previously reported.

My app requires Prev/Next buttons to scroll through the number of slides currently in view, with `skipSnaps: true` and `slidesToScroll: 1` options. I've achieved this functionality through a workaround, which involves using `slidesInView()` to calculate the number of visible slides and then scrolling to the appropriate slide. You can view my implementation here: https://stackblitz.com/edit/github-pf8j7x?file=app.vue.

I believe that incorporating this feature directly into the `scrollNext` and `scrollPrev` methods would significantly streamline this use case, enhancing the library's usability and flexibility.

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues
- [X] I agree to follow this project's Contributing Guidelines for feature requests

**Comment by davidjerleke** [comment]
@meirroth why not just use [`scrollTo()`](https://www.embla-carousel.com/api/methods/#scrollto)? 

```ts
emblaApi.value.scrollTo(desiredIndex);
```

It takes you to the exact index you want.

**Reply by meirroth** [reply]
@davidjerleke I am using [`scrollTo()`](https://www.embla-carousel.com/api/methods/#scrollto):
```javascript
function scrollPrev() {
  if (emblaApi.value && slidesInView.value.length) {
    const scrollTo = slidesInView.value[0] - slidesInView.value.length
    emblaApi.value.scrollTo(scrollTo)
  }
}

function scrollNext() {
  if (emblaApi.value && slidesInView.value.length) {
    const scrollTo = slidesInView.value[slidesInView.value.length - 1] + 1
    emblaApi.value.scrollTo(scrollTo)
  }
}
```
I'm proposing a simpler way to achieve this use case with a new param for `scrollNext` and `scrollPrev`, which will look like this:
```javascript
function scrollPrev() {
  if (emblaApi.value) {
    emblaApi.value.scrollPrev({ slidesToScroll: slidesInView.value.length })
  }
}

function scrollNext() {
  if (emblaApi.value) {
    emblaApi.value.scrollNext({ slidesToScroll: slidesInView.value.length })
  }
}
```

**Reply by davidjerleke** [reply]
@meirroth would you mind explaining what you want to do in words? Because I’m not sure I understand what you want to achieve?

**Reply by meirroth** [reply]
@davidjerleke Sure. 
I want the prev/next controls to scroll to the next set of slides, so if currently slides [1, 2, 3, 4] are shown, clicking next should scroll to slides [5, 6, 7, 8]. I want to achieve this functionality while keeping the scroll by dragging and wheel gestures free to scroll 1 at a time (`slidesToScroll: 1`).
I achieved this behavior using `scrollTo`, you can see working implementation [here](https://stackblitz.com/edit/github-pf8j7x?file=app.vue). 
Adding a new param `slidesToScroll` to `scrollNext` and `scrollPrev` will simplify the logic.

**Reply by davidjerleke** [reply]
@meirroth thanks for clarifying. I don't think we should add that to the `scrollNext()` and `scrollPrev()` method because they're there to scroll to adjacent slides. I would rather create a new method and name it something like `scrollBy()`:

```tsx
function scrollBy(snapsToScroll: number, jump: boolean) {}
```

However, this is the first time in many years that someone asks for this feature. This doesn't mean that someone else haven't thought about this of course, but at least we can say that no dev made the effort to create a feature request for this up until now. So I would like to wait and see if this feature request gets any traction and if it does, consider implementing it then. The reason for this is simply because if I were to implement all features that devs request, this library would be huge. At the time of writing, devs really appreciate Embla being very lightweight with given its size it has a high feature density, while also being very extensible. I have to think about that when implementing new stuff. I hope you understand.

With that said, you could create your own `scrollBy()` like this:

```tsx
function scrollBy(snapsToScroll, jump = false) {
  const target = emblaApi.internalEngine().index.add(snapsToScroll).get()
  emblaApi.scrollTo(target, jump, Math.sign(snapsToScroll) * -1)
}
```

And usage:

```tsx
// Scroll 4 slides forward
scrollBy(4)

// Scroll 4 slides backward
scrollBy(-4)
```

I think that should work. Haven't tested though, famous last words 😅. 

Best,
David

**Reply by meirroth** [reply]
@davidjerleke That makes sense! Thank you
